### PR TITLE
Update libfmt to 4.0.0

### PR DESCRIPTION
**Pull request by candrews**
## Description

Update Kodi to use libfmt 4.0.0 https://github.com/fmtlib/fmt/releases/tag/4.0.0
IMHO, this change is better than #12389 but it requires two additional changes, so perhaps #12389 should be merged first, then this when the prerequisites have been addressed.
Before this change can be merged, these two things must be done first:
1. http://mirrors.kodi.tv/build-deps/sources must have libfmt-4.0.0 added to it so tools/depends/target/libfmt/Makefile can get it
2. team-xbmc/xbmc-ppa-build-depends must have libfmt4-dev added and libfmt updated to version 4

FYI, [Debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=866338) and [Fedora](https://bugzilla.redhat.com/show_bug.cgi?id=1465701) requests for updating to libfmt 4.0.0

## Motivation and Context

It's always nice to use the latest versions  of things :)

## How Has This Been Tested?

I tested it on my system, but note that CI does not currently pass (see above).

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by Montellese** [comment]
We also need to update the win32 package not just the linux one(s).

**Comment by Montellese** [comment]
Looks like libfmt added a `fmt::join()` implementation which we could use to replace our custom `StringUtils::Join()`.

**Comment by Rechi** [comment]
What is the actual benefit from forcing fmt 4 instead of giving the possibility to choose between fmt 3 and 4?

**Review by Rechi — DISMISSED** [review]
fmt version for windows is bumped in `FindFmt.cmake`, but the windows fmt packages aren't updated.

File: .travis.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ install:
   - if [[ "$TRAVIS_OS_NAME" == "linux" && "$BUILD" == "Kodi" ]]; then
       sudo apt-get install -qq automake autopoint build-essential cmake curl default-jre gawk gdb gdc
       gettext git-core gperf libasound2-dev libass-dev libbz2-dev libcap-dev libcdio-dev libcec4-dev libcrossguid-dev libcurl3
-      libcurl4-openssl-dev libdbus-1-dev libfmt3-dev libfontconfig-dev libegl1-mesa-dev libfreetype6-dev libfribidi-dev libgif-dev
+      libcurl4-openssl-dev libdbus-1-dev libfmt4-dev libfontconfig-dev libegl1-mesa-dev libfreetype6-dev libfribidi-dev libgif-dev
```
Comment (wsnipex) [inline .travis.yml]: this package does not exist

**Comment by wsnipex** [comment]
does this provide anything useful, besides requiring me to package libfmt4?

**Comment by Paxxi** [comment]
Haven't looked at the actual benefits but I do like bumping our depends and keeping us up to date in the early stages of a new version.

**Comment by Rechi** [comment]
I'm not against bumping the version for depends, but I'm against bumping the minimum required version of the library if it isn't necessary.

File: .travis.yml (hunk 1 above)
Comment (candrews) [inline .travis.yml]: I know - I pointed that out in the description :)

**Comment by wsnipex** [comment]
please remove the required version bump to 4.0 as well as the readme changes related to it.
As long as 3.0 works, we will support it.

**Comment by Rechi** [comment]
@candrews will you do the requested changes?

**Comment by Rechi** [comment]
_bump_

File: docs/README.linux [hunk 2]
```diff
@@ -48,7 +48,7 @@ Build-Depends: autoconf, automake, autopoint, autotools-dev, cmake, curl,
   libass-dev (>= 0.9.8), libavahi-client-dev, libavahi-common-dev, libbluetooth-dev,
   libbluray-dev (>= 0.7.0), libbz2-dev, libcap-dev,
   libcdio-dev, libcec-dev, libcurl4-openssl-dev | libcurl4-gnutls-dev | libcurl-dev,
-  libcwiid-dev, libdbus-1-dev, libegl1-mesa-dev, libfmt3-dev, libfontconfig-dev, libfreetype6-dev,
+  libcwiid-dev, libdbus-1-dev, libegl1-mesa-dev, libfmt4-dev, libfontconfig-dev, libfreetype6-dev,
```
Comment (Rechi) [inline docs/README.linux]: `libfmt-dev | libfmt3-dev`

**Pull request closed without merging**